test/test_puma_server.rb

```diff
@@ -1141,4 +1141,12 @@ def test_client_quick_close_no_lowlevel_error_handler_call
     sleep 0.5
     assert_empty @events.stdout.string
   end
+
+  def test_run_stop_thread_safety
+    100.times do
```

Comment:
How long does this take to run?

Reply:
I believe 0.3 sec on JRuby, less on MRI.  I should have reviewed this earlier, but...

I suspect I can revert the band-aids for JRuby (the sleep after run).  This is the same issue of `Server#run` getting messed up.

Reply:
[0.02s](https://github.com/puma/puma/runs/1276840378#step:8:356) on MRI, [0.28s](https://github.com/puma/puma/runs/1276840522#step:9:422) on JRuby